### PR TITLE
[JIT] Also fold NaiveSyncBatchNorm when folding batch norm

### PR DESCRIPTION
**Pull request by SS-JIA**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#57823 [JIT] Also fold NaiveSyncBatchNorm when folding batch norm**

Some models use the `NaiveSyncBatchNorm` instead of `BatchNorm2d`, but during inference they behave the same. This change is to ensure that `NaiveSyncBatchNorm` gets folded into convs during optimization passes, particularly `FoldConvBatchNorm`.

Differential Revision: [D28291709](https://our.internmc.facebook.com/intern/diff/D28291709)

### CODE REVIEW

**Comment by SS-JIA** [comment]
@SS-JIA has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28291709).

**Review by jerryzh168 — APPROVED** [review]

**Comment by SS-JIA** [comment]
@SS-JIA has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28291709).
